File: Lab12_TextEditor/Lab12_TextEditor/EditorMemento.cs
namespace Lab12_TextEditor;

public class EditorMemento
{
    private readonly string _text;
    private readonly int _cursorPosition;
    private readonly string _selection;
    
    public EditorMemento(string text, int cursorPosition, string selection)
    {
        _text = text;
        _cursorPosition = cursorPosition;
        _selection = selection;
    }
    
    public string GetText()
    {
        return _text;
    }
    
    public int GetCursorPosition()
    {
        return _cursorPosition;
    }
    
    public string GetSelection()
    {
        return _selection;
    }
}

File: lab12/src/EditorMemento.java
public class EditorMemento {
    private String content;
    private int position;
    private String selection;

    public EditorMemento(String content, int p, String s)
    {
        this.content=content;
        this.position=p;
        this.selection=s;
    }
    public String getContent()
    {
        return content;
    }

    public int getPosition(){
        return position;
    }
    public String getSelection()
    {
        return selection;
    }


}
